src/modules/aws-ec2/iam-instance-profile/removeIamInstanceProfile.ts

```diff
@@ -0,0 +1,24 @@
+import { EC2 } from "aws-sdk";
+import { logger } from "../../../logger";
+import { getIamInstanceProfileAssociations } from "./getIamInstanceProfileAssociations";
+
+export const removeIamInstanceProfile = async (instanceId: string) => {
+  const ec2 = new EC2();
+  try {
+    const responseData = await getIamInstanceProfileAssociations(instanceId);
+    if (!responseData.IamInstanceProfileAssociations) {
+      return;
+    }
+    const profileAssociations = responseData.IamInstanceProfileAssociations;
+    profileAssociations.forEach(async association => {
+      const associationId = association.AssociationId || "";
+      const params = {
+        AssociationId: associationId
+      };
+      await ec2.disassociateIamInstanceProfile(params).promise();
+    });
+  } catch (error) {
+    logger.error(error.message);
```

Comment:
We should swallow this error here, especially considering that we would then just log the removal of the instance profile below as a success. Consider re-throwing or not putting a try/catch block at all.